src/GraphQL/Types/Scalars/GuidGraphType.cs

```diff
@@ -9,35 +9,23 @@ namespace GraphQL.Types
     public class GuidGraphType : ScalarGraphType
     {
         /// <inheritdoc/>
-        public override bool CanParseLiteral(IValue value) => value switch
-        {
-            GuidValue _ => true,
-            StringValue s => Guid.TryParse(s.Value, out _),
-            _ => false
-        };
+        public override bool CanParseLiteral(IValue value) => value is StringValue s && Guid.TryParse(s.Value, out _);
 
         /// <inheritdoc/>
-        public override bool CanParseValue(object value) => value switch
-        {
-            Guid _ => true,
-            string s => Guid.TryParse(s, out _),
-            _ => false
-        };
+        public override bool CanParseValue(object value) => value is Guid || value is string s && Guid.TryParse(s, out _);
 
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value) => value switch
-        {
-            GuidValue g => g.Value,
-            StringValue s => Guid.Parse(s.Value),
-            _ => null
-        };
+        public override object ParseLiteral(IValue value) => value is StringValue s ? Guid.Parse(s.Value) : (Guid?)null;
 
         /// <inheritdoc/>
         public override object ParseValue(object value) => value switch
         {
-            Guid _ => value, // no boxing
```

Comment:
???

Reply:
I didn't think about this.  We should of course restore these conditions to prevent boxing.

Reply:
I deliberately left an explanatory comment. I'm surprised you didn't notice it.

Reply:
Yeah... 🤷🏻‍♂️

Reply:
Fixed